presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -199,11 +198,7 @@ public HiveWriterFactory(
         this.maxOpenSortFiles = maxOpenSortFiles;
         this.sortedWritingTempStagingPathEnabled = isTemporaryStagingDirectoryEnabled(session);
         this.sortedWritingTempStagingPath = getTemporaryStagingDirectoryPath(session);
-        this.immutablePartitions = immutablePartitions;
-        this.insertExistingPartitionsBehavior = HiveSessionProperties.getInsertExistingPartitionsBehavior(session);
-        if (immutablePartitions) {
-            checkArgument(insertExistingPartitionsBehavior != InsertExistingPartitionsBehavior.APPEND, "insertExistingPartitionsBehavior cannot be APPEND");
-        }
+        this.insertExistingPartitionsBehavior = getInsertExistingPartitionsBehavior(session);
```

Comment:
do we call `validate` in `getInsertExistingPartitionsBehavior`? if not then we should

Reply:
@kokosing could you please check this conversation? 
https://github.com/prestosql/presto/pull/4999#discussion_r479471965 
I'd be glad to hear your opinion about it.